hazelcast-client-legacy/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java

```diff
@@ -148,6 +154,17 @@ private void registerInvocation(ClientInvocation clientInvocation) {
         }
     }
 
+    private void checkMaxAllowedReached() {
+        if (maxAllowedConcurrentInvocations == Integer.MAX_VALUE) {
+            return;
+        }
+        long currentConcurrentInvocations = callIdIncrementer.get() - tailCallId.get();
+        if (maxAllowedConcurrentInvocations <= currentConcurrentInvocations) {
+            throw new HazelcastOverloadException("Max allowed concurrent invocations is reached : "
```

Comment:
Yes, it is already like that. I am incrementing the tailCallId as soon as I get incoming message. And this is not correct  place if you are searching for it. It is incremented in deregisterCallId method.